compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -356,51 +353,17 @@ private class InteropPointerInputModifier<T : Component>(
             // Do not redispatch the event if it originally from this interop view.
             return
         }
-        val containerPoint = SwingUtilities.convertPoint(root, e.point, componentInfo.component)
-        val component = SwingUtilities.getDeepestComponentAt(
-            componentInfo.component,
-            containerPoint.x,
-            containerPoint.y
-        )
+        val component = getDeepestComponentForEvent(componentInfo.component, e)
         if (component != null) {
-            val componentPoint = SwingUtilities.convertPoint(root, e.point, component)
-            component.dispatchEvent(e.copy(component, componentPoint))
+            component.dispatchEvent(SwingUtilities.convertMouseEvent(e.component, e, component))
```

Comment:
Do we have a test for this logic? If we don't have, let's write it. I assume it will be `ComposeWindow` + `SwingPanel` + `window.dispatchEvent` + `panel.addMouseListener`